rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -2435,11 +2437,25 @@ public void schedulerChangeJobPriorityByValue(String sessionId, String jobId, St
             throws RestException, JobAlreadyFinishedRestException {
         try {
             Scheduler s = checkAccess(sessionId, PATH_JOBS + jobId + "/priority/byvalue" + priorityValue);
-            s.changeJobPriority(jobId, JobPriority.findPriority(Integer.parseInt(priorityValue)));
-        } catch (SchedulerException e) {
-            throw RestException.wrapExceptionToRest(e);
+            JobPriority newPriority = Arrays.stream(JobPriority.values())
```

Comment:
a more elegant solution would be to add a findPriorityExact static method in JobPriority that throws an IllegalArgumentException if not found

Reply:
Ok i'll implement this.